helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/ConstraintBasedAlgorithm.java

```diff
@@ -75,15 +75,13 @@ public OptimalAssignment calculate(ClusterModel clusterModel) throws HelixRebala
         computeOverallClusterRemainingCapacity(nodes);
 
     // Create a wrapper for each AssignableReplica.
-    Set<AssignableReplicaWithScore> toBeAssignedReplicas =
-        clusterModel.getAssignableReplicaMap().values().stream()
-            .flatMap(replicas -> replicas.stream())
-            .map(replica -> new AssignableReplicaWithScore(replica, clusterModel))
-            .collect(Collectors.toSet());
+    List<AssignableReplicaWithScore> toBeAssignedReplicas =
+        clusterModel.getAssignableReplicaMap().values().stream().flatMap(Collection::stream).map(
+            replica -> new AssignableReplicaWithScore(replica, clusterModel,
```

Comment:
Is it possible to avoid creating the object for faster performance? i.e. can we just calculate the score here with a function?

Reply:
Since it only contributes to 3% of total latency, removing this may not be worth the effort at this moment. We should keep a mental note of this.